shotover-proxy/tests/redis_int_tests/basic_driver_tests.rs

```diff
@@ -1476,3 +1572,46 @@ async fn run_all(connection: &mut Connection) {
     test_ping_echo(connection).await;
     test_time(connection).await;
 }
+
+struct Flusher {
```

Comment:
Another option for this could be integrating it into `ShotoverManager` and then passing that around instead of the Redis connection. That way we could remove some duplicate functionality.

Would have to rework lots of `ShotoverManager` though, so I'll let you decide if you want to do that. Otherwise this is fine.

Reply:
I'm trying to imagine what such an API would look like.

```rust
let shotover_manager = ShotoverManager::from_topology_file("example-configs/redis-multi/topology.yaml");
// Store connections internally within shotover_manager: 6379 is shotovers redis port and 1111, 1112, 1113 are the ports of the actual redis instances.
shotover_manager.redis_connections_async(6379, &[1111, 1112, 1113]).await; 
run_test_case(&shotover_manager);
    
fn run_test_case(shotover_manager: &ShotoverManager) {
   shotover_manager.flush();
   shotover_manager.shotover_redis_connection.set("foo", "bar");
   for connection in shotover_manager.direct_redis_connections {
      assert_eq!(connection.get("foo"), "bar");
   }
}
```

I do quite like the idea of this but I'm worried of the extra statefulness within ShotoverManager that this introduces.
I think the simplified and more flexible API would be worth it though?

Something this doesnt account for is secondary shotover connections which can be useful for some testcases.
Maybe we can just externally manage that connection for those cases?

Reply:
>secondary shotover connections

That could work in a similar way to the `shotover_redis_connection` field you showed in your example above. They could be created lazily, reusing the port we originally specified to connect to Shotover.
```rust
let shotover_manager = ShotoverManager::from_topology_file("example-configs/redis-multi/topology.yaml");
// Store connections internally within shotover_manager: 6379 is shotovers redis port and 1111, 1112, 1113 are the ports of the actual redis instances.
shotover_manager.redis_connections_async(6379, &[1111, 1112, 1113]).await; 
run_test_case(&shotover_manager);
    
fn run_test_case(shotover_manager: &ShotoverManager) {
    shotover_manager.flush();
    shotover_manager.shotover_redis_connection(0).unwrap().set("foo", "bar"); // get our first shotover connection with the index 0
    shotover_manager.shotover_redis_connection(1).unwrap().set("foo", "bar"); // get our second shotover connection with the index 1
   for connection in shotover_manager.direct_redis_connections {
      assert_eq!(connection.get("foo"), "bar");
   }
}
```

Reply:
After thinking about it I would like to land the Flusher as is and raise an issue to investigate combining with ShotoverManager.

My reasoning is:
It will touch a lot of things and so is out of scope of this PR and would therefore need to be a prereq or postreq. Making it a postreq has the advantage of allowing us to directly compare the ShotoverManager and Flusher approaches.
Additionally if the ShotoverManager approach doesnt pan out we can always leave it as the Flusher approach.

Reply:
I agree with that. Let's not overengineer it, if we end up needing the abstraction we'll do it.